.travis/shellchecker

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+#
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2019 NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Wrapper for running `shellcheck` over projects.
+
+shopt -s extglob
+cd "$(dirname "$0")/../" || exit 1
+
+# find shell files under the specified directory
+find_files () {
+    FPATH="$1"
+    grep --color=never -rl '^#\!.*bash' "${FPATH}" | sed 's/^\.\///' &
```

Comment:
I am not aware that the `-r` option can take an argument? The `man grep` page suggests that it doesn't.

Do we actually want this line to run in the background? Do we have a `wait` for this process?

Reply:
I was puzzling over that (backgrounding) too!

Reply:
Changed to `-R`.